### ISSUE TITLE
Not working on windows with mingw.

### ISSUE THREAD

**Issue by CxdInitial** [daniel5151/libc_alloc#1]
Undefined `posix_memalign`, maybe use `_aligned_malloc` instead?

![screen](https://user-images.githubusercontent.com/25840869/100564472-1deab300-32fc-11eb-930f-eae53c914195.png)

**Comment by daniel5151** [comment]
Ahhh, Windows...

I took a look at https://github.com/rust-lang/rust/blob/master/library/std/src/sys/windows/alloc.rs, and it seems like the "right" approach is to use `HeapAlloc` directly.

Alternatively, a quick fix could be to use `#[cfg(windows)]` and `#[not(cfg(windows))]` to switch between `_aligned_malloc` `posix_memalign` and respectively.

Unfortunately, I don't have a Windows development environment set up, so this isn't something I can easily test, so I'm not sure when i'll get around to fixing this issue. That said, if you're up for it, it shouldn't be too difficult to make a PR that adds windows support (using either of the two methods listed above), and it would certainly be much appreciated :)

**Comment by CxdInitial** [comment]
The way to fix this problem is simple, but is it ok to use native Windows system call? After all, this crate is call `libc_alloc`, not `native_alloc`. 

`malloc` suits the name "libc" better. Another but, `malloc` has no alignment guarantee.

**Comment by daniel5151** [comment]
Funny enough, I had originally considered calling this library `malloc_alloc`, as I did in-fact initually use `malloc` directly (until realizing that it didn't fulfill Rust's alignment guarantees). When I switched over to `posix_memalign`, I decided to go up an abstraction layer and call the crate `libc_alloc`, though in hindsignt, maybe I should have gone even more generic and called the crate `system_alloc` right off the bat. Ah well ¯\\\_(ツ)_/¯

As such, I'm not too squeamish about using non portable libc functions (case in point - `posix_memalign`).
